### PR TITLE
Support standalone executable in gz-msgs11

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 New feature

Reworks #353 for `ionic`

## Summary

This adds support for a standalone gz-utils::cli based executable for message introspection.

It additionally modifies the ruby scripts so that gz-tools will continue to work.

This also removes the `compiled` component introduced in #339, as it's not really necessary with the new standalone executable structure.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: core/cmd/msgs_main.cc [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <string>
```
Comment (ahcorde) [inline core/cmd/msgs_main.cc:18]:
```suggestion
#include <iostream>
#include <memory>
#include <string>
```

File: core/cmd/msgs_main.cc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  app.add_flag_callback("-v,--version", [](){
+      std::cout << GZ_MSGS_VERSION_FULL << std::endl;
+      throw CLI::Success();
+  });
+
+  addMsgFlags(app);
+  app.formatter(std::make_shared<GzFormatter>(&app));
+  CLI11_PARSE(app, argc, argv);
+}
```
Comment (ahcorde) [inline core/cmd/msgs_main.cc:135]:
```suggestion
    return 0;
}
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by azeey** [comment]
The description says "Reworks https://github.com/gazebosim/gz-msgs/pull/353 for harmonic", but the PR is targeting `main`. Is that right?

**Comment by mjcarroll** [comment]
> The description says "Reworks https://github.com/gazebosim/gz-msgs/pull/353 for harmonic", but the PR is targeting main. Is that right?

It should be for `main`, so ionic at this point.

File: core/cmd/msgs_main.cc (hunk 1 above)
Comment (ahcorde) [reply core/cmd/msgs_main.cc:18]: did you add the header?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: core/cmd/msgs_main.cc (hunk 1 above)
Comment (mjcarroll) [reply core/cmd/msgs_main.cc:19]: Whoops, bad merge, thanks.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**